fastly/base_fastly_service_v1.go

```diff
@@ -0,0 +1,482 @@
+			newVersion, err := conn.CloneVersion(&gofastly.CloneVersionInput{
+				Service: d.Id(),
+				Version: latestVersion,
+			})
+			if err != nil {
+				return err
+			}
+
+			// The new version number is named "Number", but it's actually a string
+			latestVersion = newVersion.Number
+			d.Set("cloned_version", latestVersion)
+
+			// New versions are not immediately found in the API, or are not
+			// immediately mutable, so we need to sleep a few and let Fastly ready
+			// itself. Typically, 7 seconds is enough
+			log.Print("[DEBUG] Sleeping 7 seconds to allow Fastly Version to be available")
+			time.Sleep(7 * time.Second)
+
+			// Update the cloned version's comment
+			if d.Get("version_comment").(string) != "" {
+				opts := gofastly.UpdateVersionInput{
+					Service: d.Id(),
+					Version: latestVersion,
+					Comment: d.Get("version_comment").(string),
+				}
+
+				log.Printf("[DEBUG] Update Version opts: %#v", opts)
+				_, err := conn.UpdateVersion(&opts)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		for _, a := range serviceDef.GetAttributeHandler() {
+			if a.MustProcess(d, initialVersion) {
+				if err := a.Process(d, latestVersion, conn); err != nil {
+					return err
+				}
+			}
+		}
+
+		// validate version
+		log.Printf("[DEBUG] Validating Fastly Service (%s), Version (%v)", d.Id(), latestVersion)
+		valid, msg, err := conn.ValidateVersion(&gofastly.ValidateVersionInput{
+			Service: d.Id(),
+			Version: latestVersion,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error checking validation: %s", err)
+		}
+
+		if !valid {
+			return fmt.Errorf("[ERR] Invalid configuration for Fastly Service (%s): %s", d.Id(), msg)
+		}
+
+		shouldActivate := d.Get("activate").(bool)
+		if shouldActivate {
+			log.Printf("[DEBUG] Activating Fastly Service (%s), Version (%v)", d.Id(), latestVersion)
+			_, err = conn.ActivateVersion(&gofastly.ActivateVersionInput{
+				Service: d.Id(),
+				Version: latestVersion,
+			})
+			if err != nil {
+				return fmt.Errorf("[ERR] Error activating version (%d): %s", latestVersion, err)
+			}
+
+			// Only if the version is valid and activated do we set the active_version.
+			// This prevents us from getting stuck in cloning an invalid version
+			d.Set("active_version", latestVersion)
+		} else {
+			log.Printf("[INFO] Skipping activation of Fastly Service (%s), Version (%v)", d.Id(), latestVersion)
+			log.Print("[INFO] The Terraform definition is explicitly specified to not activate the changes on Fastly")
+			log.Printf("[INFO] Version (%v) has been pushed and validated", latestVersion)
+			log.Printf("[INFO] Visit https://manage.fastly.com/configure/services/%s/versions/%v and activate it manually", d.Id(), latestVersion)
+		}
+	}
+
+	return resourceServiceRead(d, meta, serviceDef)
+}
+
+// resourceServiceRead provides service resource Read functionality
+func resourceServiceRead(d *schema.ResourceData, meta interface{}, serviceDef ServiceDefinition) error {
```

Comment:
Same as above, at this level, we could possibly use `conn FastlyClient` instead of `meta interface{}`.